### DISCUSSION TITLE
ccxt okx python, set_leverage doesn't work. please help me

### DISCUSSION THREAD

**Discussion by jupiter100** [community/community#52685] [Programming Help]
I want to trade using python in okx. I want to enter a position with a set leverage. However, the leverage setting is not working. Is there any way to solve this problem? Thank you in advance.

``````````````````````````````````````````````````````````````````````````````````
okex = ccxt.okex5({
    'apiKey': okx_apikey,
    'secret': okx_secretkey,
    'password': okx_password,
    'options': {
        'defaultType': 'swap'
    }
})

Tickers = okex.fetch_tickers()

symbol = 'SLP/USDT:USDT'
side = 'buy'
type = 'market'
amount = 10

posMode = okex.set_position_mode(hedged=False)
params = {"mgnMode" : "isolated", "posSide" : "long"}
okex.set_leverage(5, symbol, params)
okex.create_order(symbol, type, side, amount)
````````````````````````````````````````````````````````````````````````````````````````````

I tried print. Then the output was as follows.

```
========== set position mode ==========
{'code': '0', 'data': [{'posMode': 'net_mode'}], 'msg': ''}
========== set leverage ==========
{'code': '0',
 'data': [{'instId': 'SLP-USDT-SWAP',
           'lever': '5',
           'mgnMode': 'isolated',
           'posSide': 'long'}],
 'msg': ''}
========== market buy order ==========
{'amount': None,
 'average': None,
 'clientOrderId': 'e847386590ce4dBC403ca6bf6a011dbf',
 'cost': None,
 'datetime': None,
 'fee': None,
 'fees': [],
 'filled': None,
 'id': '566577006755155968',
 'info': {'clOrdId': 'e847386590ce4dBC403ca6bf6a011dbf',
          'ordId': '566577006755155968',
          'sCode': '0',
          'sMsg': 'Order placed',
          'tag': 'e847386590ce4dBC'},
 'lastTradeTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': False,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'SLP/USDT:USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'market'}
```

I wrote the python code like this and ran it. However, on the okx exchange, it is displayed as Cross, Buy 3.00x. What I want is Isolated, Buy 5.00x. Did I write the wrong code? please help me thank you

In 2022, someone else asked this same question. I tried following the comments on the question, but all failed. I also thought about whether okx discriminates against setting leverage according to the amount an individual holds. Because I have less than 10,000 dollars.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by sergeyleschev** [comment]
This issue might be related to the API version you are using. I suggest checking if your API version supports the set_leverage() function and if you have the proper permissions to execute it. Additionally, you could try setting the isolated margin mode before calling set_leverage(). If the problem persists, I recommend contacting the OKEx support team for further assistance.

**Reply by jupiter100** [reply]
> This issue might be related to the API version you are using. I suggest checking if your API version supports the set_leverage() function and if you have the proper permissions to execute it. Additionally, you could try setting the isolated margin mode before calling set_leverage(). If the problem persists, I recommend contacting the OKEx support team for further assistance.

Thank you for your interest in my problem. I created an API in April 2023, it is a V5 API. Could you please tell me how I can check if my API version supports the set_leverage() function and if I have proper permission to execute it? thank you

**Comment by jupiter100** [comment]
create_order -> params -> Add 'tdMode' : 'isolated'

**Reply by jupiterr-chen** [reply]
This problem has troubled me for a long time, this works for me.
